File: 087.LeetCode@230_Kth_Smallest_Element_in_a_BST/087.LeetCode@230_Kth_Smallest_Element_in_a_BST.py
class Solution:
    def inorder(self, root, ans):
        if not root:
            return

        self.inorder(root.left, ans)
        ans.append(root.val)
        self.inorder(root.right, ans)

    def kthSmallest(self, root, k):
        ans = []
        self.inorder(root, ans)
        return ans[k - 1]

File: 087.LeetCode@230_Kth_Smallest_Element_in_a_BST/087.LeetCode@230_Kth_Smallest_Element_in_a_BST.java
class Solution {
    int c = 0, ans = 0;
    public int kthSmallest(TreeNode root, int k) {
        if(root == null) {
            return 0;
        }
        kthSmallest(root.left,k);
        c++;
        if(k == c) {
            ans = root.val;
        }
        kthSmallest(root.right,k);
        return ans;
    }
}